docs-rtd/howto/manage-offers.md

```diff
@@ -78,14 +90,44 @@ resource "juju_integration" "this" {
 > See more: [`juju_integration` > `via`](https://registry.terraform.io/providers/juju/juju/latest/docs/resources/integration#via)
 
 
-## Remove an offer
-> Who: User with [offer `admin` access](https://juju.is/docs/juju/user-permissions#heading--offer-admin).
+(manage-access-to-an-offer)=
+## Manage access to an offer
 
-To remove an offer, in your Terraform plan, remove its resource definition.
+When using Juju alone, you can grant offer access only to a user. When using Juju with JAAS, you can grant offer access to a user, a group, or a service account.
 
-> See more: [`juju_offer`](https://registry.terraform.io/providers/juju/juju/latest/docs/resources/offer)
+### Using Juju alone
+When using Juju alone, to grant a user access to an offer, in your Terraform plan add a `juju_access_offer` resource, specifying the offer URL and setting the Juju access level to the list of users you want to grant that level. For example:
+
+```terraform
+resource "juju_access_offer" "this" {
+  offer_url = juju_offer.my_application_offer.url
+  consume   = [juju_user.dev.name]
+}
+```
 
+> See more: [`juju_access_offer`](https://registry.terraform.io/providers/juju/juju/latest/docs/resources/access_offer), [Juju | Offer access levels](https://canonical-juju.readthedocs-hosted.com/en/latest/user/reference/user/#valid-access-levels-for-application-offers)
 
-<br>
 
-> <small>**Contributors:** @anvial, @cderici, @hmlanigan, @manadart, @simonrichardson, @tmihoc </small>
+### Using Juju + JAAS
+When using Juju with JAAS, to grant one or more users, groups, and/or service accounts access to a model, in your Terraform plan add a resource type `juju_jaas_access_offer`, specifying the offer URL, the access level, and the desired list desired users, groups, and/or service accounts. For example:
+
+```terraform
+resource "juju_jaas_access_offer" "development" {
+  offer_url        = juju_offer.myoffer.url
+  access           = "consumer"
+  users            = ["foo@domain.com"]
+  groups           = [juju_jaas_group.development.uuid]
+  service_accounts = ["Client-ID-1", "Client-ID-2"]
+}
+```
+
+> See more: [`juju_jaas_access_offer`](https://registry.terraform.io/providers/juju/juju/latest/docs/resources/jaas_access_offer), [JAAS | Offer access levels](https://canonical-jaas-documentation.readthedocs-hosted.com/en/latest/reference/authorisation_model/#application-offer)
+
+
+
+## Remove an offer
+> Who: User with [offer `admin` access](https://canonical-juju.readthedocs-hosted.com/en/latest/user/reference/user/#user-access-offer-admin).
+
+To remove an offer, in your Terraform plan, remove its resource definition.
```

Comment:
unless they used the offer data source in which case will need to use external tools to remove it

Reply:
As discussed in our call, I've updated the docs to not have any mention of data sources in the Remove an x sections. Also, for all entities that can be referenced through a data source, I've updated the Manage x > Reference... section to have the better phrasing we agreed upon ("outside of the current Terraform plan").